Lib/extractor/formats/opentype.py

```diff
@@ -1046,3 +1049,44 @@ def _extractOpenTypeKerningFromKern(source):
         # there are no minimum values.
         kerning.update(subtable.kernTable)
     return kerning
+
+# -----
+# Color
+# -----
+
+COLOR_LAYERS_KEY = "com.github.googlei18n.ufo2ft.colorLayers"
+COLOR_PALETTES_KEY = "com.github.googlei18n.ufo2ft.colorPalettes"
+COLOR_LAYER_MAPPING_KEY = "com.github.googlei18n.ufo2ft.colorLayerMapping"
+UFO2FT_FILTER_KEY = "com.github.googlei18n.ufo2ft.filters"
+def extractColorsFromOpenType(
+    source,
+    destination,
+):
+    if not "CPAL" in source or not "COLR" in source: return
+    destination.lib[UFO2FT_FILTER_KEY] = [{"name": "Explode Color Layer Glyphs", "pre": True}]
+    # UFO format only supports one palette. Others ignored.
+    firstPalette = source["CPAL"].palettes[0]
+    colorIDs = list()
+    for i, color in enumerate(firstPalette):
+        colorIDs.append((color.red/255.0, color.green/255.0, color.blue/255.0, color.alpha/255.0))
+
+    destination.lib[COLOR_PALETTES_KEY] = [[(color.red/255.0, color.green/255.0, color.blue/255.0, color.alpha/255.0) for color in firstPalette]]
+
+    layerGlyphs = list()
+    for (base_glyph, layer_records) in source["COLR"].ColorLayers.items():
```

Comment:
this is assuming COLR v0 table (flat colors only); maybe you should check the table.version and error out nicely if it's == 1 (instead of AttributeError).
[COLRv1](https://github.com/googlefonts/colr-gradients-spec/) defines a tree of paint tables with support for gradients and much more. It's not officially in OpenType yet so it's ok if extractor doesn't support this until finalized.